### PR TITLE
feat: add dbconfig package to abstract the generation of database config

### PR DESCRIPTION
**Pull request by zyy17**

### CODE REVIEW

File: pkg/dbconfig/datanode_config.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	if cluster.Spec.Datanode != nil && len(cluster.Spec.Datanode.Config) > 0 {
+		if err := Merge([]byte(cluster.Spec.Datanode.Config), c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Kind returns the component kind of the frontend.
```
Comment (daviderli614) [inline pkg/dbconfig/datanode_config.go:160]: kind of the datanode.

File: pkg/dbconfig/datanode_config.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+	accessKeyID = s3Credentials.Data[AccessKeyIDSecretKey]
+	if accessKeyID == nil {
+		err = fmt.Errorf("secret '%s/%s' does not have key '%s'", namespace, name, AccessKeyIDSecretKey)
+		return
+	}
+
+	secretAccessKey = s3Credentials.Data[SecretAccessKeySecretKey]
+	if secretAccessKey == nil {
+		err = fmt.Errorf("secret '%s/%s' does not have key '%s'", namespace, name, SecretAccessKeySecretKey)
```
Comment (daviderli614) [inline pkg/dbconfig/datanode_config.go:190]: suggestion：Same  error message, we should change it.

File: pkg/dbconfig/datanode_config.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	"github.com/GreptimeTeam/greptimedb-operator/pkg/utils"
+)
+
+var _ Config = &DatanodeConfig{}
+
+// DatanodeConfig is the configuration for the datanode.
+type (
+	DatanodeConfig struct {
+		// Node running mode.
+		Mode string `toml:"mode,omitempty"`
```
Comment (daviderli614) [inline pkg/dbconfig/datanode_config.go:32]: question: [initializer](https://github.com/GreptimeTeam/greptimedb-operator/blob/develop/cmd/initializer/types.go#L35) is define `DatanodeConfig` struct, now here define too, what kind of behavior will there be?

File: pkg/dbconfig/datanode_config.go (hunk 1 above)
Comment (zyy17) [reply pkg/dbconfig/datanode_config.go:160]: good catch.

File: pkg/dbconfig/datanode_config.go (hunk 3 above)
Comment (zyy17) [reply pkg/dbconfig/datanode_config.go:32]: The initializer will be refactored to use `dbconfig` package. The original configs will be removed. We put it in the next [PR](https://github.com/GreptimeTeam/greptimedb-operator/pull/100).

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**